baseapp/abci_utils_test.go

```diff
@@ -415,6 +416,93 @@ func (s *ABCIUtilsTestSuite) TestDefaultProposalHandler_NoOpMempoolTxSelection()
 	}
 }
 
+func (s *ABCIUtilsTestSuite) TestDefaultProposalHandler_PriorityNonceMempoolTxSelection() {
+	cdc := codectestutil.CodecOptions{}.NewCodec()
+	baseapptestutil.RegisterInterfaces(cdc.InterfaceRegistry())
+	txConfig := authtx.NewTxConfig(cdc, authtx.DefaultSignModes)
+	ctrl := gomock.NewController(s.T())
+	app := mock.NewMockProposalTxVerifier(ctrl)
```

Comment:
Why do we need a mock controller for this test?

Reply:
Because I need to mock this function “txBz, err := h.txVerifier.PrepareProposalVerifyTx(memTx)”
just mock it to reply like this “app.EXPECT().PrepareProposalVerifyTx(tx4).Return(txBz4, nil).AnyTimes()”